addons/project/controllers/main.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+
+
+class ProjectController(http.Controller):
+
+    @http.route('/project/plan/<int:project_id>', type='http', auth='user', methods=['GET'])
+    def plan(self, project_id):
+        project = request.env['project.project'].browse(project_id)
+        values = self._prepare_plan_values(project)
+        return request.render('project.project_plan', values)
+
+    def _prepare_plan_values(self, project):
+        user_id = request.env.user.id
+        values = {
+            'project': project,
+        }
+        if project.use_tasks:
+            values.setdefault('todo', {}).update({
```

Comment:
Dafuq? Why use setdefault then update it? the `values` dict is created de-novo two lines above, why not just `values['todo'] = {…}`?